app/javascript/utilities/http/__tests__/csrfToken.test.js

```diff
@@ -0,0 +1,14 @@
+import { getCSRFToken } from '@utilities/http/csrfToken';
+
+describe('getCSRFToken', () => {
+  it('should resolve with the CSRF token when there is a meta tag', async () => {
+    document.head.innerHTML = `<meta name="csrf-token" content="this-is-a-csrf-token"></head>`;
+
+    const data = await getCSRFToken();
+    expect(data).toBe('this-is-a-csrf-token');
+  });
+
+  // [@ridhwana] It was proving to be really difficult to test the error case
```

Comment:
This feels like a silly question, but is it actually possible for it to fail? I guess we could break the layout, fail to render the `meta` tag somehow, but that seems like a Rails-side failure. Maybe there are other scenarios I'm not thinking of?

Reply:
No, thats basically the error case that I was trying to replicate is when the promise gets rejected because there's no csrf token in the meta tag. That does seem like a rails side failure, do you think we don't need to test this within the test?

Reply:
I'm generally a big fan of graceful degradation, but when it's hard-to-impossible to simulate the failure case, I think there's a kind of diminishing returns in spending the effort to create the test.